util/upload-cookbook.py

```diff
@@ -0,0 +1,241 @@
+
+        raise e
+
+
+def _aws_bck(s3, args, region, bucket_name, full_name):
+    if args.dryrun:
+        print(
+            "Not backing up {0} to bucket {1} override is {2}, dryrun is {3}".format(
+                full_name, bucket_name, args.override, args.dryrun
+            )
+        )
+    else:
+        try:
+            copy_source = {"Bucket": bucket_name, "Key": _COOKBOOKS_DIR + "/" + full_name}
+            s3.copy(copy_source, bucket_name, _BACKUP_DIR + "/" + full_name + _bck_date)
+        except ClientError as e:
+            print("Couldn't backup {0}".format(full_name))
+            if e.response["Error"]["Code"] == "NoSuchBucket":
+                print("Bucket is not present.")
+            _bck_error_array.add(region)
+
+
+def _aws_cp(s3, args, region, bucket_name, folder, full_name):
+    if args.dryrun:
+        print(
+            "Not uploading {0} to bucket {1}, override is {2}, dryrun is {3}".format(
+                full_name, bucket_name, args.override, args.dryrun
+            )
+        )
+    else:
+        key = folder + "/" + full_name
+        try:
+            s3.upload_file(full_name, bucket_name, key, ExtraArgs={"ACL": "public-read"})
+
+            print("Successfully uploaded {0} to s3://{1}/{2}".format(full_name, bucket_name, key))
+        except ClientError as e:
+            print("Couldn't upload {0} to bucket s3://{1}/{2}".format(full_name, bucket_name, key))
+            _cp_error_array.add(region)
+            if e.response["Error"]["Code"] == "NoSuchBucket":
+                print("Bucket is not present.")
+
+            raise e
+
+
+def _create_s3_client(region):
+    reg_credentials = [c for c in credentials if c[0] == region]
+
+    if reg_credentials:
+        credential = reg_credentials[0]
+        credential_region = credential[0]
+        credential_endpoint = credential[1]
+        credential_arn = credential[2]
+        credential_external_id = credential[3]
+
+        try:
+            sts = boto3.client("sts", region_name=main_region, endpoint_url=credential_endpoint)
+
+            assumed_role_object = sts.assume_role(
+                RoleArn=credential_arn,
+                ExternalId=credential_external_id,
+                RoleSessionName=credential_region + "upload_cfn_templates_sts_session",
+            )
+            aws_credentials = assumed_role_object["Credentials"]
+            s3 = boto3.client(
+                "s3",
+                region_name=credential_region,
+                aws_access_key_id=aws_credentials.get("AccessKeyId"),
+                aws_secret_access_key=aws_credentials.get("SecretAccessKey"),
+                aws_session_token=aws_credentials.get("SessionToken"),
+            )
+
+        except ClientError:
+            print("Warning: non authorized in region '{0}', skipping".format(credential_region))
+            pass
+    else:
+        s3 = boto3.client("s3")
+    return s3
+
+
+def _get_bucket_name(args, region):
```

Comment:
I'd use bucket=Null as first parameter in place of args that is a more complex structure
```python
def _get_bucket_name(region, bucket=Null):

# then you can call it with:
_get_bucket_name(region, args.bucket)
```